plugin/src/main/java/org/opensearch/ml/processor/MLInferenceSearchResponseProcessor.java

```diff
@@ -125,9 +126,16 @@ public SearchResponse processResponse(SearchRequest request, SearchResponse resp
     /**
      * Processes the search response asynchronously by rewriting the documents with the inference results.
      *
-     * @param request         the search request
-     * @param response        the search response
-     * @param responseContext the pipeline processing context
+     * At default, process many document in one prediction through rewriteResponseDocuments method.
```

Comment:
at default, when one_to_one is false,

we run rewriteResponseDocuments by send a response that has N hits, meaning N document, and it will compile a list of model input with N documents, for example, when N=2, the model input is `"input":["document1", "document2"]`

 when one_to_one is true, 
Step 1:  in processResponseAsync method, it decompose the ONE response with N hits into  N responses with one hit, https://github.com/opensearch-project/ml-commons/blob/b52053980615afe1167ac7c458978d26a2e70215/plugin/src/main/java/org/opensearch/ml/processor/MLInferenceSearchResponseProcessor.java#L170 and

Step 2:  run the same rewriteResponseDocuments method, in this way, it sends N model input with one document/hit. For example, when N=2, the first model input `"input":"document1"`, the second model input `"input": "document2"`, after every response aded the model predictions into the hits. 

Step 3: The combineResponseListener will combine the N responses with one hit back into ONE response with N hits, and the return the ONE response.

Reply:
summarized the java doc, is it better? @dhrubo-os 

```
    /**
     * Processes the search response asynchronously by rewriting the documents with the inference results.
     *
     * By default, it processes multiple documents in a single prediction through the rewriteResponseDocuments method.
     * However, when processing one document per inference, it separates the N-hits search response into N one-hit search responses,
     * executes the same rewriteResponseDocument method for each one-hit search response,
     * and after receiving N one-hit search responses with inference results,
     * it combines them back into a single N-hits search response.
     *
     * @param request          the search request
     * @param response         the search response
     * @param responseContext  the pipeline processing context
     * @param responseListener the listener to be notified when the response is processed
     */
```

Reply:
That's better. Thanks.

Reply:
changed in the new commit https://github.com/opensearch-project/ml-commons/pull/2801/commits/eca844b5850c502a8acd207c2f7a0d442839bd8f